scripts/DP28-JP/c100428032.lua

```diff
@@ -0,0 +1,88 @@
+--BK アッパーカッター
+
+--Script by Chrono-Genex
+function c100428032.initial_effect(c)
+	--to hand
+	local e1=Effect.CreateEffect(c)
+	e1:SetDescription(aux.Stringid(100428032,0))
+	e1:SetCategory(CATEGORY_TOHAND+CATEGORY_SEARCH)
+	e1:SetType(EFFECT_TYPE_SINGLE+EFFECT_TYPE_TRIGGER_O)
+	e1:SetProperty(EFFECT_FLAG_DELAY)
+	e1:SetCode(EVENT_SUMMON_SUCCESS)
+	e1:SetCountLimit(1,100428032)
+	e1:SetTarget(c100428032.thtg)
+	e1:SetOperation(c100428032.thop)
+	c:RegisterEffect(e1)
+	local e2=e1:Clone()
+	e2:SetCode(EVENT_SPSUMMON_SUCCESS)
+	c:RegisterEffect(e2)
+	--leave grave
+	local e3=Effect.CreateEffect(c)
+	e3:SetDescription(aux.Stringid(100428032,1))
+	e3:SetType(EFFECT_TYPE_SINGLE+EFFECT_TYPE_TRIGGER_O)
+	e3:SetProperty(EFFECT_FLAG_DELAY)
+	e3:SetCode(EVENT_TO_GRAVE)
+	e3:SetCountLimit(1,100428032)
+	e3:SetCondition(c100428032.lgcon)
+	e3:SetTarget(c100428032.lgtg)
+	e3:SetOperation(c100428032.lgop)
+	c:RegisterEffect(e3)
+end
+function c100428032.thfilter(c)
+	return (c:IsSetCard(0x1084) and c:IsType(TYPE_MONSTER) and not c:IsCode(100428032)
+		or c:IsSetCard(0x299) and c:IsType(TYPE_COUNTER)) and c:IsAbleToHand()
+end
+function c100428032.thtg(e,tp,eg,ep,ev,re,r,rp,chk,_,exc)
+	if chk==0 then return Duel.IsExistingMatchingCard(c100428032.thfilter,tp,LOCATION_DECK,0,1,exc) end
+	Duel.SetOperationInfo(0,CATEGORY_TOHAND,nil,1,tp,LOCATION_DECK)
+end
+function c100428032.thop(e,tp,eg,ep,ev,re,r,rp)
+	Duel.Hint(HINT_SELECTMSG,tp,HINTMSG_ATOHAND)
+	local g=Duel.SelectMatchingCard(tp,c100428032.thfilter,tp,LOCATION_DECK,0,1,1,nil)
+	if g:GetCount()>0 then
+		Duel.SendtoHand(g,nil,REASON_EFFECT)
+		Duel.ConfirmCards(1-tp,g)
+	end
+end
+function c100428032.lgcon(e,tp,eg,ep,ev,re,r,rp)
+	return e:GetHandler():IsReason(REASON_EFFECT)
+end
+function c100428032.spfilter(c,e,tp)
+	return c:IsSetCard(0x1084) and c:IsType(TYPE_MONSTER) and not c:IsCode(100428032) and c:IsCanBeSpecialSummoned(e,0,tp,false,false)
+end
+function c100428032.setfilter(c)
+	return c:IsSetCard(0x299) and c:IsType(TYPE_COUNTER) and c:IsSSetable()
+end
+function c100428032.lgtg(e,tp,eg,ep,ev,re,r,rp,chk)
+	local b1=Duel.GetLocationCount(tp,LOCATION_MZONE)>0
+		and Duel.IsExistingMatchingCard(c100428032.spfilter,tp,LOCATION_GRAVE,0,1,nil,e,tp)
+	local b2=Duel.IsExistingMatchingCard(c100428032.setfilter,tp,LOCATION_GRAVE,0,1,nil)
+	if chk==0 then return b1 or b2 end
+	local op=0
+	if b1 and b2 then op=Duel.SelectOption(tp,aux.Stringid(100428032,1),aux.Stringid(100428032,2))
+	elseif b1 then op=Duel.SelectOption(tp,aux.Stringid(100428032,1))
+	else op=Duel.SelectOption(tp,aux.Stringid(100428032,2))+1 end
```

Comment:
Any particular reason option index 0 uses the same string ID as at https://github.com/Fluorohydride/ygopro-pre-script/pull/1074/files#diff-2e49ef6f1f7a3ed6bf867de7dcf28220b21b32f85420006ace8baa8193a7cf10R21 ?
```suggestion
	if b1 and b2 then op=Duel.SelectOption(tp,aux.Stringid(100428032,2),aux.Stringid(100428032,3))
	elseif b1 then op=Duel.SelectOption(tp,aux.Stringid(100428032,2))
	else op=Duel.SelectOption(tp,aux.Stringid(100428032,3))+1 end
```